l10n_it_asset_management/models/asset.py

```diff
@@ -0,0 +1,251 @@
+# Author(s): Silvio Gregorini (silviogregorini@openforce.it)
+# Copyright 2019 Openforce Srls Unipersonale (www.openforce.it)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class Asset(models.Model):
+    _name = "asset.asset"
+    _description = "Assets"
+    _inherit = ["mail.thread", "mail.activity.mixin", "portal.mixin"]
+    _order = "purchase_date desc, name asc"
+
+    @api.model
+    def get_default_company_id(self):
+        return self.env.user.company_id
+
+    asset_accounting_info_ids = fields.One2many(
+        "asset.accounting.info", "asset_id", string="Accounting Info"
+    )
+
+    category_id = fields.Many2one(
+        "asset.category",
+        required=True,
+        string="Category",
+    )
+
+    code = fields.Char(
+        default="",
+    )
+
+    company_id = fields.Many2one(
+        "res.company",
+        default=get_default_company_id,
+        required=True,
+        string="Company",
+        tracking=True,
+    )
+
+    currency_id = fields.Many2one(
+        "res.currency",
+        required=True,
+        string="Currency",
+    )
+
+    customer_id = fields.Many2one("res.partner", string="Customer")
+
+    depreciation_ids = fields.One2many(
+        "asset.depreciation",
+        "asset_id",
+        string="Depreciations",
+    )
+
+    name = fields.Char(
+        required=True,
+        tracking=True,
+    )
+
+    purchase_amount = fields.Monetary(
+        string="Purchase Value",
+        tracking=True,
+    )
+
+    purchase_date = fields.Date(
+        default=fields.Date.today(),
+        tracking=True,
+    )
+
+    purchase_move_id = fields.Many2one("account.move", string="Purchase Move")
+
+    sale_amount = fields.Monetary(
+        string="Sale Value",
+    )
+
+    sale_date = fields.Date()
+
+    sale_move_id = fields.Many2one("account.move", string="Sale Move")
+
+    sold = fields.Boolean()
+
+    state = fields.Selection(
+        [
+            ("non_depreciated", "Non Depreciated"),
+            ("partially_depreciated", "Partially Depreciated"),
+            ("totally_depreciated", "Depreciated"),
+        ],
+        compute="_compute_state",
+        default="non_depreciated",
+        store=True,
+    )
+
+    supplier_id = fields.Many2one("res.partner", string="Supplier")
+
+    supplier_ref = fields.Char(string="Supplier Ref.")
+
+    used = fields.Boolean()
+
+    @api.model
+    def create(self, vals):
+        # Add depreciation if it's missing while category is set
```

Comment:
Dovresti usare il decoratore `@api.model_create_multi` e modificare la funzione per warning `is not overriding the create method in batch`

Reply:
Fatto

Reply:
ora la `create` ritorna solo un asset ma nel `create_multi` possono esserci puoi record da creare